gcapi/apibase.py

```diff
@@ -143,14 +143,20 @@ def iterate_all(self, params=None) -> Iterator[T]:
             yield from current_list
             offset += req_count
 
-    def detail(self, pk=None, **params) -> Generator[T, dict[Any, Any], T]:
+    def detail(
+        self, pk=None, api_url=None, **params
+    ) -> Generator[T, dict[Any, Any], T]:
         if all((pk, params)):
             raise ValueError("Only one of pk or params must be specified")
```

Comment:
I think that the combination of `pk` and `api_url` should also be incompatible. Shouldn't we also check that only one of `pk` or `api_url` or `params` are set?

Reply:
Ah yes, I forgot to manually copy this along! But it should really check that only one of them is set.